### PR TITLE
NodeJS Week2 Homework Meal-sharing

### PR DESCRIPTION
**Pull request by slastushevskyi**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -2,13 +2,93 @@ const express = require("express");
...
 
-router.get("/", async (request, response) => {
+// GET	Returns all meals
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const data = await knex("Meal").select();
+    res.status(200).json(await data);
```
Comment (itsjrm) [inline src/backend/api/meals.js]: At this point `data` has been resolved in the line above. You do not have to await it again.

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -2,13 +2,93 @@ const express = require("express");
...
+        if (value === requestedId) {
+          return meal;
+        }
+      }
+    });
+    if (filteredMeal.length === 0) {
+      res.status(404).end("Meal Id Not Found");
+    } else {
+      const [{ id }] = filteredMeal;
+      const meal = await knex("Meal").select().where("id", id);
```
Comment (itsjrm) [inline src/backend/api/meals.js]: Would it make sense to only make this call? We could check the response and determine if the `id` was found or not. 
Maybe something like:
```
try {
    const requestedId = parseInt(req.params.id);
    const meal = await knex("Meal").select().where("id", requestedId);
    if(!meal.length) {
      res.status(404).end(`Meal Id: ${requestedId} Not Found`);
    }
    res.status(200).json({ data: meal });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
```

File: src/backend/api/reservations.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    console.error(error);
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+});
+
+// POST Adds a new reservation to the database
+router.post("/", async (req, res) => {
+  try {
+    const newReservation = req.body;
+    const [id] = await knex("Reservation").insert(newReservation);
```
Comment (itsjrm) [inline src/backend/api/reservations.js]: This is a [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) of the returned array, This means that you have the actual value of the id in a const called `id`.
Therefore, for the check on the next line you can simply put in `if(id)`

**Review by itsjrm — CHANGES_REQUESTED** [review]
Nice work on this weeks Meal-Sharing assignment!
I left you a few comments. Please let me know if something needs clarifying.

Happy coding!

File: src/backend/api/meals.js (hunk 2 above)
Comment (slastushevskyi) [reply src/backend/api/meals.js]: Agree. This is much easier to read and way shorter!

File: src/backend/api/reservations.js (hunk 3 above)
Comment (slastushevskyi) [reply src/backend/api/reservations.js]: It's hard to remember why i did that, but i remember that reason was that knex.raw returns array in array. Probably i deleted the part of code that was connected with it somehow 🤷‍♂️🙆‍♂️

Comment (slastushevskyi) [reply src/backend/api/reservations.js]: One more time, Jane, thank you for your review. I'm appreciate that. Hope. you will help me by reviewing my NodeJS week 3 homework🤝🙏

Comment (itsjrm) [reply src/backend/api/reservations.js]: My pleasure!

**Pull request merged**